### PR TITLE
RAS-872: Extend EQ launch claims to include sds_dataset_id if required for the form_type

### PR DESCRIPTION
**Pull request by SteveScorfield**
# What and why?
The SDS ID is required to be passed to eQ when an SDS is present and the form type(s) in question match that supplied by SDS. This PR achieves this by grabbing the SDS attributes from the database and passing this to Fronstage for further processing.

# How to test?
1. Deploy this PR to your local environment
2. Add a CE in rOps and then add an SDS that matches the CE - this can be done by either directly adding it to the DB or adding it to the pubsub subscription 'supplementary-data-service-subscription-[YOUR-ENV]'.
3. Hit the endpoint: '/collectionexercises/[EXERCISE-ID]' of the collection exercise service
4. The SDS values should be present within the 'supplementaryDatasetJson' node of the returned JSON

Note: You can also test this when an SDS is not present. The 'supplementaryDatasetJson' will be null.

# Jira
https://jira.ons.gov.uk/browse/RAS-872

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java [hunk 1]
```diff
@@ -66,4 +66,12 @@ private SupplementaryDatasetEntity createSupplementaryDatasetEntity(
           e);
     }
   }
+
+  public SupplementaryDatasetEntity findSupplementaryDataset(int exercise_pk) {
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java:70]: rename to exercise_pk to exercisePk

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/domain/CollectionExercise.java [hunk 2]
```diff
@@ -93,4 +95,9 @@ public class CollectionExercise {
 
   @Column(name = "eq_version")
   private String eqVersion;
+
+  //  @OneToOne(optional = true) is true by default
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/domain/CollectionExercise.java:99]: delete commented code

**Review by JacobHarrisonONS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/domain/CollectionExercise.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/domain/CollectionExercise.java:99]: Done

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java (hunk 1 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java:70]: Done

**Pull request merged**